camera/EmulatedCameraDevice.cpp

```diff
@@ -285,22 +285,48 @@ bool EmulatedCameraDevice::inWorkerThread()
 
 status_t EmulatedCameraDevice::WorkerThread::readyToRun()
 {
+    Mutex::Autolock _l(mLock);
     ALOGV("Starting emulated camera device worker thread...");
 
     ALOGW_IF(mThreadControl >= 0 || mControlFD >= 0,
             "%s: Thread control FDs are opened", __FUNCTION__);
     /* Create a pair of FDs that would be used to control the thread. */
     int thread_fds[2];
+    status_t rv = NO_ERROR;
     if (pipe(thread_fds) == 0) {
         mThreadControl = thread_fds[1];
         mControlFD = thread_fds[0];
         ALOGV("Emulated device's worker thread has been started.");
-        return NO_ERROR;
     } else {
         ALOGE("%s: Unable to create thread control FDs: %d -> %s",
-             __FUNCTION__, errno, strerror(errno));
-        return errno;
+              __FUNCTION__, errno, strerror(errno));
```

Comment:
It looks like this line picked up an extra leading space. If this was unintentional, removing it will make the diff a little cleaner.